packages/sdk-v2/src/sdk/savings/react.ts

```diff
@@ -0,0 +1,174 @@
+import React, { useCallback } from 'react'
+import { useContractFunction, useCalls, QueryParams } from '@usedapp/core'
+import { useGetContract } from '../base/react'
+import { ethers } from 'ethers'
+import { EnvKey } from '../base'
+import { GoodDollarStaking, IGoodDollar } from '@gooddollar/goodprotocol/types'
+import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
+import { G$ } from '@gooddollar/web3sdk'
+
+export interface StakerInfo {
+  claimable: {
+    g$Reward: CurrencyAmount<Currency> | any,
+    goodReward: CurrencyAmount<Currency> | any
+  } | undefined, 
+  balance?: CurrencyAmount<Currency> | any,
+  rewardsPaid?: {
+    g$Minted: CurrencyAmount<Currency> | any,
+    goodMinted: CurrencyAmount<Currency> | any,
+  }
+  deposit?: CurrencyAmount<Currency> | any,
+  principle?: CurrencyAmount<Currency> | any
+  shares?: number | any,
+  lastSharePrice?: number | any
+}
+
+export interface GlobalStats {
+  totalStaked: CurrencyAmount<Currency> | undefined,
+  totalRewardsPaid: CurrencyAmount<Currency> | undefined,
+  apy: number | undefined
+  lastUpdateBlock?: number,
+  savings?: number,
+}
+
+
+export const useSavingsFunctions = (env?: EnvKey) => {
+  const gooddollar = useGetContract("GoodDollar", true, "savings", env) as IGoodDollar;
+  const gdStaking = useGetContract("GoodDollarStaking", true, "savings", env) as GoodDollarStaking;
+
+  const {state: transferState, send: sendTransfer} = useContractFunction(gooddollar, "transferAndCall", {transactionName: "Transfer to savings"});
+  const {state: withdrawState, send: sendWithdraw} = useContractFunction(gdStaking, "withdrawStake", {transactionName: "withdraw from savings"});
+  const {state: claimState, send: sendClaim} = useContractFunction(gdStaking, "withdrawRewards", {transactionName: 'Withdraw rewards from savings'});
+
+  const transfer = useCallback((amount: string) => {
+    const callData = ethers.constants.HashZero
+    return sendTransfer(gdStaking.address, amount, callData)
+  }, [sendTransfer, gdStaking]);
+
+  const withdraw = useCallback(async(amount:string, address?: string) => {
+    const shares = address ? await gdStaking.sharesOf(address) : await gdStaking.amountToShares(amount) // sharesOf used to withdraw full amount
+    return sendWithdraw(shares)
+  }, [sendWithdraw]);
+
+  const claim = useCallback(() => sendClaim(), [sendClaim])
+
+  return { transfer, withdraw, claim, transferState, withdrawState, claimState }
+}
+
+export const useGlobalStats = (refresh: QueryParams["refresh"] = "never", env?: EnvKey) => {
+const gdStaking = useGetContract("GoodDollarStaking", true, "savings", env) as GoodDollarStaking;
```

Comment:
missing indentation. make sure you have prettier configured correctly

Reply:
fixed